src/Model/Union.php

```diff
@@ -0,0 +1,464 @@
+                }
+            }
+
+            // now prepare query
+            $expr[] = '[' . $cnt . ']';
+            $query = $this->persistence->action($nestedModel, 'select', [false]);
+
+            if ($nestedModel instanceof self) {
+                $subquery = $nestedModel->getSubQuery($fields);
+                //$query = parent::action($mode, $args);
+                $query->reset('table')->table($subquery);
+
+                if (isset($nestedModel->group)) {
+                    $query->group($nestedModel->group);
+                }
+            }
+
+            $query->field($queryFieldExpressions);
+
+            // also for sub-queries
+            if ($this->group) {
+                if (is_array($this->group)) {
+                    foreach ($this->group as $group) {
+                        if (isset($fieldMap[$group])) {
+                            $query->group($nestedModel->expr($fieldMap[$group]));
+                        } elseif ($nestedModel->hasField($group)) {
+                            $query->group($nestedModel->getField($group));
+                        }
+                    }
+                } elseif (isset($fieldMap[$this->group])) {
+                    $query->group($nestedModel->expr($fieldMap[$this->group]));
+                } else {
+                    $query->group($this->group);
+                }
+            }
+
+            // subquery should not be wrapped in parenthesis, SQLite is especially picky about that
+            $query->allowToWrapInParenthesis = false;
+
+            $args[$cnt++] = $query;
+        }
+
+        // last element is table name itself
+        $args[$cnt] = $this->table;
+
+        return $this->persistence->dsql()->expr('(' . implode(' UNION ALL ', $expr) . ') {' . $cnt . '}', $args);
+    }
+
+    /**
+     * No description.
+     */
+    public function getSubAction(string $action, array $act_arg = []): Expression
+    {
+        $cnt = 0;
+        $expr = [];
+        $args = [];
+
+        foreach ($this->union as [$model, $mapping]) {
+            // now prepare query
+            $expr[] = '[' . $cnt . ']';
+            if ($act_arg && isset($act_arg[1])) {
+                $a = $act_arg;
+                $a[1] = $this->getFieldExpr(
+                    $model,
+                    $a[1],
+                    $mapping[$a[1]] ?? null
+                );
+                $query = $model->action($action, $a);
+            } else {
+                $query = $model->action($action, $act_arg);
+            }
+
+            // subquery should not be wrapped in parenthesis, SQLite is especially picky about that
+            $query->allowToWrapInParenthesis = false;
+
+            $args[$cnt++] = $query;
+        }
+
+        // last element is table name itself
+        $args[$cnt] = $this->table;
+
+        return $this->persistence->dsql()->expr('(' . implode(' UNION ALL ', $expr) . ') {' . $cnt . '}', $args);
```

Comment:
should be more line `(() union all ())`

Reply:
NO PARENTHESIS !
I can be mistaken which exactly, but one of SQL engines didn't support `(A) union (B)`, should be `A union B` without parenhesis.

Reply:
fine